modules/custom/openy_daxko2/src/Form/ImportForm.php

```diff
@@ -0,0 +1,247 @@
+              $value = substr($value, 0, 10);
+              break;
+
+            case 'locations':
+              // We expect 'locations' to be overidden with hook.
+              // @see openy_daxko2_example_openy_daxko2_categories_csv_row_alter().
+
+              break;
+
+            case 'program':
+              array_pop($value);
+              $value['description'] = $row['description'];
+
+              $categories[$value['id']] = $value;
+
+              $value = $value['id'];
+              break;
+
+            case 'days_offered':
+              $weekdays = [];
+              foreach ($value as $day) {
+                $weekdays[] = $day['name'];
+              }
+              $value = implode(', ', $weekdays);
+              break;
+
+            case 'registration':
+              $value = json_encode($value);
+              break;
+
+            case 'restrictions':
+              if (isset($value['age'])) {
+                $value = json_encode($value);
+              }
+              break;
+
+            case 'times':
+              $value = json_encode([
+                'times' => $row['times'],
+                'days' => $row['days_offered'],
+                'start_date' => $row['start_date'],
+                'end_date' => $row['end_date'],
+              ]);
+              break;
+          }
+
+          $newRow[$key] = $value;
+        }
+
+        $newRow['link'] = 'https://ops1.operations.daxko.com/Online/' . $config->get('client_id') . '/ProgramsV2/OfferingDetails.mvc?program_id=' . $row['program']['id'] . '&offering_id=' . $row['id'] . '&location_id=' . $row['locations'][0]['id'];
+        $newRow['ageFrom'] = isset($row['restrictions']['age']) ? $row['restrictions']['age']['start'] : null;
+        $newRow['ageTo'] = isset($row['restrictions']['age']) ? $row['restrictions']['age']['end'] : null;
+
+        // Allow custom implementations to set up the mappings.
+        \Drupal::moduleHandler()->alter('openy_daxko2_programs_csv_row', $newRow);
+
+        fputcsv($fp, $newRow);
+      }
+
+      $after = '';
+      if (isset($programsResponse['links'])) {
+        $link = reset($programsResponse['links']);
+        if (strpos($link['href'], 'after=')) {
+          list(, $after) = explode('after=', $link['href']);
+          $after = urldecode($after);
+        }
+      }
+
+      $i++;
+    }
+
+    fclose($fp);
+
+    // Save categories CSV file.
+    unlink('/tmp/categories.csv');
+    $fp = fopen('/tmp/categories.csv', 'w');
+    foreach ($categories as $fields) {
+      \Drupal::moduleHandler()->alter('openy_daxko2_categories_csv_row', $fields);
+
+      fputcsv($fp, $fields);
+    }
+    fclose($fp);
+  }
+
+  public static function batchFinished($success, $results, $operations) {
+    // For some reason imported sessions (offerings) got status 13 instead of 1.
+    // So update them manually or dive into the code and find the bug.
+    db_query('UPDATE {node_field_data} SET status=1 WHERE status=13');
```

Comment:
Is it possible to use db_update here?
We should think about the possibility to have this code working on SQLite and MSSQL. db_query makes this impossible